Failure in test_fittable_compound

I'm running into the following error on MacOS X with Python 3.6 and Scipy 0.19:

```
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0

Running tests with Astropy version 2.0.dev18503.
Running tests in astropy/modeling/tests/test_mappings.py.

Date: 2017-06-06T11:04:44

Platform: Darwin-15.6.0-x86_64-i386-64bit

Executable: /Users/tom/miniconda3/envs/dev/bin/python

Full Python Version: 
3.6.1 |Continuum Analytics, Inc.| (default, May 11 2017, 13:04:09) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.12.1
Scipy: 0.19.0
Matplotlib: 2.0.2
h5py: 2.7.0
Pandas: 0.20.1
Cython: 0.25.2
Using Astropy options: remote_data: none.

rootdir: /private/var/folders/zy/t1l3sx310d3d6p0kyxqzlrnr0000gr/T/astropy-test-750h6_b3, inifile: setup.cfg
collected 7 items 

astropy/modeling/tests/test_mappings.py ......F

=============================================================================== FAILURES ===============================================================================
________________________________________________________________________ test_fittable_compound ________________________________________________________________________

    @pytest.mark.skipif('not HAS_SCIPY')
    def test_fittable_compound():
        m = Identity(1) | Mapping((0, )) | Gaussian1D(1, 5, 4)
        x = np.arange(10)
        y_real = m(x)
        dy = 0.005
        y_noisy = y_real + np.random.normal(0., dy, x.shape)
        pfit = LevMarLSQFitter()
        new_model = pfit(m, x, y_noisy)
        y_fit = new_model(x)
>       assert_allclose(y_fit, y_real, rtol=dy)
E       AssertionError: 
E       Not equal to tolerance rtol=0.005, atol=0
E       
E       (mismatch 10.0%)
E        x: array([ 0.46022 ,  0.608963,  0.75704 ,  0.884198,  0.970249,  1.000275,
E               0.968853,  0.881657,  0.753779,  0.605467])
E        y: array([ 0.457833,  0.606531,  0.75484 ,  0.882497,  0.969233,  1.      ,
E               0.969233,  0.882497,  0.75484 ,  0.606531])

astropy/modeling/tests/test_mappings.py:87: AssertionError
======================================================================== pytest-warning summary ========================================================================
WC1 None [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
======================================================== 1 failed, 6 passed, 1 pytest-warnings in 1.43 seconds =========================================================
```

This happens only when running this test file directly, not modeling as a whole suggesting some kind of state issue:

```
python setup.py test -t astropy/modeling/tests/test_mappings.py 
```

--- a/astropy/modeling/tests/test_mappings.py
+++ b/astropy/modeling/tests/test_mappings.py
@@ -8,6 +8,7 @@ from numpy.testing.utils import assert_allclose, assert_array_equal
 
 from ..fitting import LevMarLSQFitter
 from ..models import Shift, Rotation2D, Gaussian1D, Identity, Mapping
+from ...utils import NumpyRNGContext
 
 try:
     from scipy import optimize  # pylint: disable=W0611
@@ -80,8 +81,10 @@ def test_fittable_compound():
     x = np.arange(10)
     y_real = m(x)
     dy = 0.005
-    y_noisy = y_real + np.random.normal(0., dy, x.shape)
+    with NumpyRNGContext(1234567):
+        n = np.random.normal(0., dy, x.shape)
+    y_noisy = y_real + n
     pfit = LevMarLSQFitter()
     new_model = pfit(m, x, y_noisy)
     y_fit = new_model(x)
-    assert_allclose(y_fit, y_real, rtol=dy)
+    assert_allclose(y_fit, y_real, atol=dy)

